File: task-03/Javascript/4/Sub-Task4.js
const fs = require('fs');

function generateDiamond(n) {
    let lines = [];

    for (let i = 0; i < n; i++) {
        let spaces = ' '.repeat(n - i - 1);
        let stars = '*'.repeat(2 * i + 1);
        lines.push(spaces + stars);
    }

    for (let i = n - 2; i >= 0; i--) {
        let spaces = ' '.repeat(n - i - 1);
        let stars = '*'.repeat(2 * i + 1);
        lines.push(spaces + stars);
    }

    return lines.join('\n');
}

function main() {
    fs.readFile('input.txt', 'utf8', (err, data) => {
        if (err) {
            console.error('Error reading input file:', err);
            return;
        }
        
        let n = parseInt(data.trim(), 10);
        if (isNaN(n)) {
            console.error('Invalid number format in input file.');
            return;
        }

        let diamondPattern = generateDiamond(n);

        fs.writeFile('output.txt', diamondPattern, (err) => {
            if (err) {
                console.error('Error writing to output file:', err);
                return;
            }
            console.log('Pattern printed in output.txt');
        });
    });
}

main();

File: task-03/Haskell/4/Sub-Task4.hs
import System.IO
import Control.Monad (when)

generateDiamond :: Int -> String
generateDiamond n = unlines $ upperPart ++ lowerPart
  where
    upperPart = [ replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*' | i <- [0..n-1] ]
    lowerPart = [ replicate (n - i - 1) ' ' ++ replicate (2 * i + 1) '*' | i <- reverse [0..n-2] ]

main :: IO ()
main = do
    content <- readFile "input.txt"
    let n = read (filter (/= ' ') content) :: Int

    let diamondPattern = generateDiamond n

    writeFile "output.txt" diamondPattern

    putStrLn "Pattern printed to output.txt"
